google/ggrc-core
# GGRC-5942/GGRC-5617/GGRC-6633 Do not allow to autogenerate Tickets at Ticket Tracker for Issues and Assessments in some statuses

# Issue description

User can trigger new ticket autogeneration for assessments and issues in completed statuses. There is a restriction on Ticket Tracker API side, it allows to create tickets only in 'New' or 'Assigned' statuses. It means that, if Assessment is in 'Completed and Verified' status, the ticket linked to such assessment will have 'Assigned' status, until cron job will change it to Fixed (Verified). This can cause confusions for users, who already completed an assessment and then receive a notification from Ticket Tracker system, that it is required to complete the same assessment.

**SOLUTION:**

Do not allow to autogenerate tickets for assessments that are in 'In Review', 'Completed (no verification)', 'Completed and Verified' and 'Deprecated' statuses.

**ACCEPTANCE CRITERIA:**
**Via UI:** 

For assessments at statuses 'In Review', 'Completed (no verification)', 'Completed and Verified' and 'Deprecated':
1. Change help text near TICKET ID field to:
You are not allowed to generate new ticket for Assessments at statuses "In Review", "Completed (no verification)", "Completed and Verified" and "Deprecated", only manual linking is allowed to perform. If you would like to keep the existing ticket linked to this assessment do not edit this attribute. If you would like to link to a different ticket provide an existing ticket number.
2. Ticket ID field should become mandatory and can not be empty
3. If user clicks 'Save and Close' he should see an error near Ticket ID field (can not be blank) and no tickets autogeneration should happen
4. For assessments at other statuses use usual UI and autogeneration logic. 
5. Update `Ticket ID` modal for Ticket Tracker inline editing.

**Via import:** 

1. If user decided to import Assessments at statuses 'In Review', 'Completed (no verification)', 'Completed and Verified' and 'Deprecated'
- On Import page show Warning:
Line <X>: You are not allowed to autogenerate tickets at Ticket Tracker for Assessments at statuses 'In Review', 'Completed (no verification)', 'Completed and Verified' and 'Deprecated' statuses.  Column '<Y>' will be ignored. Please use a manual linking option instead.
where X - Number of the row at import file where the assessment at this statuses appear
Y - Number of the column "Ticket Tracker Switch" at import file 
-  User can click 'Proceed in spite of warnings' , the cell value should be ignored, no tickets autogeneration should happen for this assessment.

# Steps to test the changes

1. Try to import new assessment without status column (Default 'Draft' value should be applied) and Ticket Tracker integration turned ON. Check if ticket was generated and no warnings were occurred.
2. Try to import new assessment with status column in the In Review state and Ticket Tracker integration turned ON. Check if there were appropriate warning and ticket wasn't generated and integration was turned OFF. 
3. Create assessment via UI with integration turned OFF in 'Draft' or 'Active' status. Turn Ticket Tracker integration ON via import for this Issue. Check if ticket were generated and no warnings occurred.
4. Create issue via UI with integration turned OFF in  'In Review', 'Completed (no verification)', 'Completed and Verified' status. Turn Ticket Tracker integration ON via import for this assessment.  Check if there were appropriate warning and ticket wasn't generated and integration was turned OFF. 

# Solution description

FE: In scope of #8909 PR were lost changes for Issue. So these changes are added to this PR.
Add validation rule for ticket Id field for Assessment object.

# Sanity checklist




# PR Review checklist
